File: Android/doric/src/main/java/com/github/pengfeizhou/doric/extension/bridge/DoricPromise.java
package com.github.pengfeizhou.doric.extension.bridge;

import com.github.pengfeizhou.doric.DoricContext;
import com.github.pengfeizhou.doric.utils.DoricConstant;
import com.github.pengfeizhou.jscore.JavaValue;

/**
 * @Description: com.github.pengfeizhou.doric.extension.bridge
 * @Author: pengfei.zhou
 * @CreateDate: 2019-07-19
 */
public class DoricPromise {
    private final DoricContext context;
    private final String callbackId;

    public DoricPromise(DoricContext context, String callbackId) {
        this.context = context;
        this.callbackId = callbackId;
    }

    public void resolve(JavaValue... javaValue) {
        context.getDriver().invokeDoricMethod(
                DoricConstant.DORIC_BRIDGE_RESOLVE,
                context.getContextId(),
                callbackId,
                javaValue);
    }

    public void reject(JavaValue... javaValue) {
        context.getDriver().invokeDoricMethod(
                DoricConstant.DORIC_BRIDGE_REJECT,
                context.getContextId(),
                callbackId,
                javaValue);
    }
}

File: iOS/Pod/Classes/Plugin/DoricPromise.m
//
//  DoricPromise.m
//  Doric
//
//  Created by pengfei.zhou on 2019/7/29.
//

#import "DoricPromise.h"
#import "DoricConstant.h"

@interface DoricPromise ()
@property(nonatomic, strong) DoricContext *context;
@property(nonatomic, strong) NSString *callbackId;

@end

@implementation DoricPromise

- (instancetype)initWithContext:(DoricContext *)context callbackId:(NSString *)callbackId {
    if (self = [super init]) {
        _context = context;
        _callbackId = callbackId;
    }
    return self;
}

- (void)resolve:(id)result {
    [self.context.driver invokeDoricMethod:DORIC_BRIDGE_RESOLVE, self.context.contextId, self.callbackId, result, nil];
}

- (void)reject:(id)result {
    [self.context.driver invokeDoricMethod:DORIC_BRIDGE_REJECT, self.context.contextId, self.callbackId, result, nil];
}
@end
